### DISCUSSION TITLE
New user: can't find whether autocomplete of for instance Inner Join works

### DISCUSSION THREAD

**Discussion by raoulteeuwenPE** [microsoft/azuredatastudio#16900] [Q&A] [answered]
Hello. Sorry if this post is not in the right place etc. Just started using AZS. Love it so far. But can't find how to get auto complete working for commands like 'inner join', ' order by' etc. Is there a way, and if so, what is that way :-)? Thank you.

**Comment by gregclerk** [comment]
Do you mean like Devart's "SQL Complete" extension for SSMS where it will intelligently predict the join table and fields based off of the primary key-foreign key relationship?  If so, I'd love to have this too!

(That and auto aliasing)

**Comment by erinstellato-ms** [comment, marked as answer]
Hi @raoulteeuwenPE (and @gregclerk) - the autocomplete for joins isn't available with the standard intellisense in Azure Data Studio, but if you use the Copilot extension and you provide the DDL for the objects involved, you get much better prompting/auto-complete.  If you're not familiar, check out Drew's post: https://techcommunity.microsoft.com/t5/azure-sql-blog/introducing-github-copilot-for-azure-data-studio/ba-p/3829511

**Reply by gregclerk** [reply]
Thanks!  I'll definitely look into this!

**Comment by raoulteeuwenPE** [comment]
For some reason at some moment, simple 'ij' followed by a tab would not auto complete to INNER JOIN etc. But probably user error; it is a long time since i started using ADS, and at some point i got it working, tx, also for the more advanced options with CoPilot :-)
